Prevent padding from placeholder instructions to be overriden

thealexandrelara opened the pull request:
This PR is related to the issue https://github.com/woocommerce/woocommerce-blocks/issues/6739 in which there is a margin missing for the Product Selector description on Filter Products by Attribute block

### About the implementation
The initial idea was to force the style to be applied by using `!important` on the padding-bottom property. However, after discussing more about it, it was decided that a better alternative is to add the instructions content to the body of the Placeholder content instead of relying on the `instructions` prop of the component.

This way, instead of forcing the style to be applied which could cause some side effects in the future, we are wrapping the instructions content in an HTML element that we control and, thus, can modify the styles independently of the external library it's being used (@wordpress/components).

<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/6739

* Force padding style to be applied to the component and prevent it from being overriden by the inner component styles.

### Screenshots

| Before | After |
| ------ | ----- |
|  <img width="1490" alt="image" src="https://user-images.githubusercontent.com/20469356/199095910-55e2d6c7-e6a6-480a-a57f-cb0e593e4ffb.png"> | <img width="1490" alt="image" src="https://user-images.githubusercontent.com/20469356/199095732-bca6880a-0a5f-413c-acde-5bfc6449c20a.png"> |

### Testing

#### User Facing Testing

1. Add a “Filter Products by Attribute” block
2. Notice that the text "Display a list of filters based on a chosen attribute" is NOT too close to the separator

### Changelog

> Fix margins from Product Selector description on Filter Products by Attribute block

thealexandrelara committed:
Prevent padding from placeholder instructions to be overriden

Inside the AttributeFilter component we are using a component Placeholder from the @wordpress/components library. This component has its own styles applied to itself including the part related to the instructions that are displayed.

Currently there is no way to override the styles specifically for the instructions part so the only possible way is to force the style to be applied over the inner styles set by the Placeholder component.

sunyatasattva commented on assets/js/blocks/attribute-filter/editor.scss:
```diff
@@ -9,7 +9,7 @@
 		border-bottom: 1px solid #e0e2e6;
 		width: 100%;
-		padding-bottom: 1em;
+		padding-bottom: 1em !important;
```
It's generally a good idea to avoid `!important` statements. Was this the only way to make this work consistently? Isn't there a way to increase specificity in this CSS selector, or by adding a class within the component render function?

thealexandrelara replied:
Thank you for reviewing it @sunyatasattva. Yeah, I agree and don't like the idea of using !important as well, but as commented in the commit message, the Placeholder component comes from the @wordpress/components library and that component already has a style applied to it, so even if we add styles to our own component, it is going to be overridden by the library.

I think the only way to avoid using !important would be if the library didn't have any style applied to it or if it allowed us to modify styles for the inner components (like for the instructions inside the Placeholder and not only for the Placeholder component itself)

Do you see another way to solve this?

sunyatasattva replied:
Great commit message, by the way! I had missed this as people here aren't usually as descriptive, I love it. May I suggest that you make sure the bulk of the information in the commit message are also in the PR description? It's useful when navigating and reviewing PRs.

Also, regarding the “Fixes” there should be the number of the issue after that: such as “Fixes #6739”, in this way GitHub will link the issues in the UI and also close the issue automatically when this PR is merged.

Now, regarding your points: as mentioned in the issue, it seems that the “Featured Product” (among others) as pretty much the same UI and should use the same components, how come it does have the correct margin? I'd start from there. I quickly looked at it, and it seems that this block has an extra `legend` element. Maybe this element should also be in the other components and they are doing it wrong? But it seems the ones without the `legend` element do better.

Other solutions would be wrapping the Placeholder element into another `div` and style that one? Or something like this.

thealexandrelara replied:
Thank you @sunyatasattva! I'm going to move the information from the commit message to the PR description

> Also, regarding the “Fixes” there should be the number of the issue after that: such as “Fixes #6739”, in this way GitHub will link the issues in the UI and also close the issue automatically when this PR is merged.

I just added a number there, related to the issue (I had manually linked both before 😅 ), so I think it is linked correctly now.

> I quickly looked at it, and it seems that this block has an extra legend element. Maybe this element should also be in the other components and they are doing it wrong? But it seems the ones without the legend element do better.

I investigated as you suggested and now I'm following the same strategy that was done by the Feature Product block, so we have a div wrapping the content and that we have control and can apply the styles as needed :D

sunyatasattva reviewed (commented):
Added a small comment on the code itself. Heads up: you are missing the “Fixes #” in the description, and you've got a typo in the changelog.

thealexandrelara committed:
Add a wrapper to the instructions content and apply the styles to it

The initial idea was to force the style to be applied by using `!important` on the padding-bottom property. However, after discussing more about it, it was decided that a better alternative is to add the instructions content to the body of the Placeholder content instead of relying on the `instructions` prop of the component.

This way, instead of forcing the style to be applied which could cause some side effects in the future, we are wrapping the instructions content in an HTML element that we control and, thus, can modify the styles independently of the external library it's being used (@wordpress/components).

sunyatasattva reviewed (approved):
LGTM! Thanks for addressing the feedback!

thealexandrelara committed:
Merge branch 'trunk' into style/fix-margin-from-product-selector-description
